x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/user/TransportHasPrivilegesAction.java

```diff
@@ -58,6 +67,15 @@ protected void doExecute(Task task, HasPrivilegesRequest request, ActionListener
             return;
         }
 
+        if (request.indexPrivileges() != null) {
+            try {
+                DLSRoleQueryValidator.validateQueryField(request.indexPrivileges(), xContentRegistry);
```

Comment:
I don't think this is right. has-privileges shouldn't be doing anything with DLS...

Reply:
I think has-privileges should probably fail the request if there is DLS/FLS in the privilege descriptor.
However, I assume it doesn't do that now, so when backporting to 7.x we could make it a deprecation and not bother validating the query itself.
WDYT?